AudioDeviceExample/ViewController.swift

```diff
@@ -212,7 +308,14 @@ class ViewController: UIViewController {
          * The important thing to remember when using a custom TVIAudioDevice is that the device must be set
          * before performing any other actions with the SDK (such as creating Tracks, or connecting to a Room).
          */
-        TwilioVideo.audioDevice = ExampleCoreAudioDevice.init();
+        TwilioVideo.audioDevice = self.audioDevice
+
+        if #available(iOS 11.0, *) {
+            // Only the ExampleAudioEngineDevice supports local audio capturing.
+            if ((TwilioVideo.audioDevice as? ExampleAudioEngineDevice) != nil) {
+                localAudioTrack = TVILocalAudioTrack()
+            }
+        }
```

Comment:
I wonder if there is a cleaner way to do this that doesn't involve a class check. Let me think on it a little.

Reply:
Nice, this looks good. 👍